extmod/modmachine.c

```diff
@@ -37,7 +37,9 @@
 
 #if MICROPY_PY_MACHINE
 
+#ifndef PAGE_SIZE
 #define PAGE_SIZE 4096
+#endif
```

Comment:
Reading man for getpagesize, there's many steps how it can be done even "righter", and that's exactly why I did it KISS way like that. I just pushed fix which should avoid naming conflicts. But note that if MSYS2 doesn't provide mmap(), the modmachine shouldn't be enabled for it (won't work as expected anyway).